action/candidate_selfstake.go

```diff
@@ -0,0 +1,53 @@
+package action
+
+import (
+	"math/big"
+
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/pkg/version"
+)
+
+const (
+	// CandidateSelfStakeBaseIntrinsicGas represents the base intrinsic gas for CandidateSelfStake
+	CandidateSelfStakeBaseIntrinsicGas = uint64(10000)
+)
+
+// CandidateSelfStake is the action to update a candidate's bucket
+type CandidateSelfStake struct {
+	AbstractAction
+
+	// bucketID is the bucket index want to be changed to
+	bucketID uint64
```

Comment:
does it need to carry a valid endorsement for the calling delegate? otherwise how to make sure the bucket is intended for this delegate?

Reply:
endorsement info will be recorded in db by another transaction `CandidateEndorse`